File: compiler/GENERATOR/BYTECODE/TEST/google.java
/*
   Usage:
    1. cd googleapi; jar xf googleapi.jar; cd ..  // Why!?
    2. javac -classpath googleapi google.java
    3. java -classpath .:googleapi google
       or
       java -classpath .:googleapi google beta.net
*/ 

// import com.google.soap.search; // fails???
import java.io.*;

public class google
{
    
  public static void main(String[] arguments) 
    { 
      String licence = "hweCKhQvbXY6kvKmOpDS2P3lTqQ5ggBu";
      String query = "";
      int i;
      for (i=0; i<arguments.length; i++) query += arguments[i] + " ";
      if (query == "") {
	System.out.print("Search Google for: ");
	query = readLn();
      }
      if (query == "") {
	query = "Ole Lehrmann Madsen BETA.NET";
      }
      System.out.println("\nSearching Google for: " + query);
      com.google.soap.search.GoogleSearch s = new com.google.soap.search.GoogleSearch();
      s.setKey(licence);
      s.setQueryString(query);
      try {
	  com.google.soap.search.GoogleSearchResult r = s.doSearch();
	  System.out.println("Results:" + "\n" + r.toString() + "\n");
      } catch (com.google.soap.search.GoogleSearchFault ne) { 
      }
    }

    public static String readLn() {
        //set aside memory for the keyboard input
        byte [] byteArray = new byte[300];
        
        //things which could fail should be enclosed
        // in a try/catch statement
        try {
            // read stdin into the byte array.
            System.in.read(byteArray);
        } catch (Exception e) {
            // if there is a failure, print an error
            e.printStackTrace();
        }
        
        //make a String from the array of bytes
        String result = new String(byteArray);
	
        //check for a return character, and only
        //keep the characters up to the return
        int newLineIndex = result.indexOf("\n");
        result = result.substring(0,newLineIndex);
        
        return result;
    }

}

File: compiler/GENERATOR/BYTECODE/TEST/google.cs
/*
  Compilation statements: 
    csc -target:library -out: googleapi/dotnet/GoogleSearchService.dll googleapi/dotnet/GoogleSearchService.cs
    csc /r:System.dll /r:googleapi/dotnet/GoogleSearchService.dll google.cs 
*/ 
using System;
using System.Text.RegularExpressions;

class Google
{
  public static void Main(String[] arguments) 
    { 
      String licence = "hweCKhQvbXY6kvKmOpDS2P3lTqQ5ggBu";
      String query = "";
      int i;
      foreach (String a in arguments) query += a + " ";
      if (query == "") {
	Console.Write("Search Google for: ");
	query = Console.ReadLine();
      }
      if (query == "") {
	query = "Ole Lehrmann Madsen BETA.NET";
      }
      Console.WriteLine("\nSearching Google for: " + query);
      GoogleSearchService s = new GoogleSearchService();
      GoogleSearchResult r = s.doGoogleSearch(licence, query, 0, 10, false, "", false, "", "latin1", "latin1");
      Console.WriteLine("\n" + r.resultElements.Length + " Results:");
      for (i=0; i<r.resultElements.Length; i++){
	// Print result stripped for HTML using Regular Expressions
	Regex paras  = new Regex("<p>");     // match HTML paragraph break tags
	Regex breaks = new Regex("<br>");    // match HTML line break tags
	Regex tags   = new Regex("<[^>]+>"); // match HTML tags
	Regex less   = new Regex("&lt;");    // match HTML less
	Regex greater= new Regex("&gt;");    // match HTML greater
	Regex amp    = new Regex("&amp;");   // match HTML ampersand
	String title = r.resultElements[i].title;
	String snippet = r.resultElements[i].snippet;

	title = tags.Replace(title, "");
	Console.WriteLine("\n" + (i+1) + ". " + title);

	snippet = paras.Replace(snippet, "\n");
	snippet = breaks.Replace(snippet, "\n");
	snippet = tags.Replace(snippet, "");
	snippet = less.Replace(snippet, "<");
	snippet = greater.Replace(snippet, ">");
	snippet = amp.Replace(snippet, "&");

	Console.WriteLine(snippet);
	Console.WriteLine("URL: " + r.resultElements[i].URL);
      }
      Console.WriteLine();
    }
}






